### ISSUE TITLE
timeScale of sp.skeleton in runtime animation cache mode

### ISSUE THREAD

**Issue by septbr** [cocos/cocos-engine#17728]
### Cocos Creator version

3.8.2

### System information

window

### Issue description

在实时缓存模式下，动画播放速率 貌似等于 timeScale * timeScale
![image](https://github.com/user-attachments/assets/3007f1b3-2ee0-4b42-810c-069669c93f5d)
![image](https://github.com/user-attachments/assets/ed3f6ab7-2f74-4361-bfc6-dd5d1dded750)
![image](https://github.com/user-attachments/assets/1028db18-e0f1-4836-b1c5-99c6594b3813)

### Relevant error log output

speed: 0.2
expected time: 7.5
frame time: 36.90009999999404
track time: 1.50053071975708

speed: 0.5
expected time: 3
frame time: 6.01630000000447
track time: 1.504073143005371

speed: 1
expected time: 1.5
frame time: 1.5163999999985098
track time: 1.5164010524749756

speed: 2
expected time: 0.75
frame time: 0.3832000000029802
track time: 1.5328000783920288

speed: 4
expected time: 0.375
frame time: 0.10009999999403954
track time: 1.601599931716919

speed: 8
expected time: 0.1875
frame time: 0.0335
track time: 2.1440000534057617

### Steps to reproduce

```typescript
import { game } from 'cc';
import { _decorator, Component, sp } from 'cc';
const { ccclass, property } = _decorator;

@ccclass('Xtk')
export class Xtk extends Component {
    @property(sp.Skeleton) private skeleton: sp.Skeleton = null;
    protected start() {
        this.test([0.2, 0.5, 1, 2, 4, 8]);
    }

    private test(speeds: number[]) {
        let speed = speeds.shift();
        if (!speed) return;

        let t0 = game.frameStartTime;
        let animation = this.skeleton.findAnimation("Skill");
        this.skeleton.timeScale = speed;
        this.skeleton.setAnimationCacheMode(sp.AnimationCacheMode.REALTIME);
        this.skeleton.setCompleteListener(track => {
            console.log("\nspeed:", speed);
            console.log("expected time:", animation.duration / speed);
            console.log("frame time:", (game.frameStartTime - t0) / 1000);
            console.log("track time:", track.trackTime);
            this.test(speeds);
        });
        this.skeleton.setAnimation(0, animation.name, false);
    }
}

```

**Comment by bofeng-song** [comment]
@septbr 该[PR](https://github.com/cocos/cocos-engine/pull/16745) 已经修复了，请合并对应修改
或升级到384

**Comment by septbr** [comment]
&nbsp;


iCunda
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: ***@***.***&gt;; 
发送时间: 2024年10月22日(星期二) 中午1:43
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [cocos/cocos-engine] timeScale of sp.skeleton in runtime animation cache mode (Issue #17728)














 
Closed #17728 as completed.
